admin auth fix

@claude

we've been going around this issue in circles for a while now - I've opened up a new issue so you can analyze this properly - please ensure that any solution you do doesn't cause rippling errors and double check that no new errors will occur.

2025-10-19T20:25:58.054550537Z ==> Downloading cache...
2025-10-19T20:25:58.056511009Z ==> Cloning from https://github.com/muppetbrown/mana_meeples_singles_market
2025-10-19T20:25:58.588179201Z ==> Checking out commit 9800933f2845330e30a4108518a0ea7ac886508c in branch main
2025-10-19T20:26:10.141964628Z ==> Transferred 80MB in 8s. Extraction took 3s.
2025-10-19T20:26:13.295171444Z ==> Requesting Node.js version 20.x
2025-10-19T20:26:13.573215843Z ==> Using Node.js version 20.19.5 via environment variable NODE_VERSION
2025-10-19T20:26:13.598687875Z ==> Docs on specifying a Node.js version: https://render.com/docs/node-version
2025-10-19T20:26:13.660973842Z ==> Running build command 'corepack enable && pnpm install --no-frozen-lockfile && pnpm run build:render'...
2025-10-19T20:26:14.273373061Z Scope: all 3 workspace projects
2025-10-19T20:26:14.792049973Z 
2025-10-19T20:26:14.792066743Z devDependencies: skipped because NODE_ENV is set to production
2025-10-19T20:26:14.792068963Z 
2025-10-19T20:26:14.808796653Z Done in 1s
2025-10-19T20:26:15.215560955Z 
2025-10-19T20:26:15.215579205Z > mana-meeples-singles-market@ build:render /opt/render/project/src
2025-10-19T20:26:15.215583285Z > pnpm --filter ./apps/web run build && pnpm --filter ./apps/api run build && node scripts/copy-dist-to-api.js
2025-10-19T20:26:15.215587235Z 
2025-10-19T20:26:15.690606851Z 
2025-10-19T20:26:15.690632331Z > web@0.1.0 build /opt/render/project/src/apps/web
2025-10-19T20:26:15.690636881Z > vite build
2025-10-19T20:26:15.690639941Z 
2025-10-19T20:26:15.890003597Z vite v5.4.20 building for production...
2025-10-19T20:26:16.203268365Z transforming...
2025-10-19T20:26:19.519608342Z ✓ 1621 modules transformed.
2025-10-19T20:26:19.667672008Z rendering chunks...
2025-10-19T20:26:19.881230803Z computing gzip size...
2025-10-19T20:26:19.888428489Z dist/index.html                             1.08 kB │ gzip:  0.55 kB
2025-10-19T20:26:19.88847103Z dist/assets/index-Bpst_Zeg.css             45.15 kB │ gzip:  7.88 kB
2025-10-19T20:26:19.888576762Z dist/assets/requestThrottler-CPgOJH9g.js    1.29 kB │ gzip:  0.55 kB
2025-10-19T20:26:19.888627813Z dist/assets/AdminLogin-QVDHDNNF.js          4.52 kB │ gzip:  1.87 kB
2025-10-19T20:26:19.888667383Z dist/assets/VirtualCardGrid-CBbuRk4j.js     4.76 kB │ gzip:  2.00 kB
2025-10-19T20:26:19.888708814Z dist/assets/CurrencySelector-HWXcCKek.js    5.62 kB │ gzip:  1.99 kB
2025-10-19T20:26:19.888756015Z dist/assets/AdminDashboard-CgWWz0oV.js     49.91 kB │ gzip: 10.81 kB
2025-10-19T20:26:19.888779945Z dist/assets/api-DEX6NwYu.js                55.89 kB │ gzip: 13.35 kB
2025-10-19T20:26:19.888894617Z dist/assets/TCGShop-AtFwNrKD.js            82.68 kB │ gzip: 21.72 kB
2025-10-19T20:26:19.888938527Z dist/assets/index-yexgffQF.js             169.41 kB │ gzip: 55.43 kB
2025-10-19T20:26:19.889160441Z ✓ built in 3.97s
2025-10-19T20:26:20.343476682Z 
2025-10-19T20:26:20.343504012Z > @mana-meeples/api@ build /opt/render/project/src/apps/api
2025-10-19T20:26:20.343508962Z > tsc -p tsconfig.json
2025-10-19T20:26:20.343511802Z 
2025-10-19T20:26:23.452361053Z src/routes/auth.ts(103,43): error TS2345: Argument of type 'any[]' is not assignable to parameter of type '[chunk: any, encoding: BufferEncoding, cb?: (() => void) | undefined]'.
2025-10-19T20:26:23.452412894Z   Target requires 2 element(s) but source may have fewer.
2025-10-19T20:26:23.480969936Z /opt/render/project/src/apps/api:
2025-10-19T20:26:23.481009737Z  ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @mana-meeples/api@ build: `tsc -p tsconfig.json`
2025-10-19T20:26:23.481013307Z Exit status 2
2025-10-19T20:26:23.515172489Z  ELIFECYCLE  Command failed with exit code 2.
2025-10-19T20:26:23.547280199Z ==> Build failed 😞
2025-10-19T20:26:23.547294629Z ==> Common ways to troubleshoot your deploy: https://render.com/docs/troubleshooting-deploys

--- a/apps/api/src/routes/auth.ts
+++ b/apps/api/src/routes/auth.ts
@@ -95,7 +95,7 @@ const debugResponse = (req: Request, res: Response, next: express.NextFunction) 
 
   // Override res.end to catch any direct response endings
   const originalEnd = res.end.bind(res);
-  res.end = function(...args: any[]) {
+  res.end = function(...args: Parameters<typeof res.end>) {
     console.log(`🔍 END: res.end called with ${args.length} args:`, args);
     logResponseState("before end()");
 

